src/storage/raw/raw_mvcc.rs

```diff
@@ -0,0 +1,215 @@
+        Ok(RawMvccIterator::new(self.snap.iter_cf(cf, iter_opt)?))
+    }
+
+    #[inline]
+    fn lower_bound(&self) -> Option<&[u8]> {
+        self.snap.lower_bound()
+    }
+
+    #[inline]
+    fn upper_bound(&self) -> Option<&[u8]> {
+        self.snap.upper_bound()
+    }
+
+    fn ext(&self) -> S::Ext<'_> {
+        self.snap.ext()
+    }
+}
+
+pub struct RawMvccIterator<I: Iterator> {
+    inner: I,
+    cur_key: Option<Vec<u8>>,
+    cur_value: Option<Vec<u8>>,
+    is_valid: Option<bool>,
+}
+
+fn is_user_key_eq(left: &[u8], right: &[u8]) -> bool {
+    let len = left.len();
+    if len != right.len() {
+        return false;
+    }
+    Key::is_user_key_eq(left, &right[..len - U64_SIZE])
+}
+
+impl<I: Iterator> RawMvccIterator<I> {
+    fn new(inner: I) -> Self {
+        RawMvccIterator {
+            inner,
+            cur_key: None,
+            cur_value: None,
+            is_valid: None,
+        }
+    }
+
+    fn update_cur_kv(&mut self) {
+        if let Some(ref mut key) = self.cur_key {
+            key.clear();
+            key.extend_from_slice(self.inner.key());
+        } else {
+            self.cur_key = Some(Vec::from(self.inner.key()));
+        };
+        if let Some(ref mut value) = self.cur_value {
+            value.clear();
+            value.extend_from_slice(self.inner.value());
+        } else {
+            self.cur_value = Some(Vec::from(self.inner.value()));
+        };
+        self.is_valid = Some(true);
+    }
+
+    fn clear_cur_kv(&mut self) {
+        self.cur_key = None;
+        self.cur_value = None;
+        self.is_valid = None;
+    }
+
+    fn move_to_prev_max_ts(&mut self) -> Result<bool> {
+        if self.inner.valid()? {
+            self.update_cur_kv();
+            self.inner.prev()?;
+        } else {
+            self.clear_cur_kv();
+            return Ok(false);
+        }
+        while self.inner.valid()? {
+            // cur_key should not be None here.
+            if is_user_key_eq(self.cur_key.as_ref().unwrap(), self.inner.key()) {
+                self.update_cur_kv();
+                self.inner.prev()?;
+            } else {
+                break;
+            }
+        }
+        Ok(true)
+    }
+}
+
+impl<I: Iterator> Iterator for RawMvccIterator<I> {
+    fn next(&mut self) -> Result<bool> {
+        let cur_key = self.inner.key().to_owned();
+        let mut res = self.inner.next()?;
+        while res && self.inner.valid()? && is_user_key_eq(&cur_key, self.inner.key()) {
+            res = self.inner.next()?;
+        }
+        self.clear_cur_kv();
+        Ok(res)
+    }
+
+    fn prev(&mut self) -> Result<bool> {
+        self.move_to_prev_max_ts()
+    }
+
+    fn seek(&mut self, key: &Key) -> Result<bool> {
+        self.clear_cur_kv();
+        self.inner.seek(key)
+    }
+
+    fn seek_for_prev(&mut self, key: &Key) -> Result<bool> {
+        if self.inner.seek_for_prev(key)? {
+            self.move_to_prev_max_ts()
```

Comment:
This is confusing as `seek` doesn't require moving to max_ts.

Reply:
seek_for_prev also need move to max_ts. As seek_for_prev reach the end the user key, which is the min ts written.

Reply:
rawkv get interface now do not accept timestamp parameter and always seek the lastest one for user.

Reply:
Since it's a public interface, its behavior should better be well documented.

Reply:
ok, comment is added to clear this.